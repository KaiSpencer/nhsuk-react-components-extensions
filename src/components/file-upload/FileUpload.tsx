import { Hint } from 'nhsuk-react-components';
import React, { HTMLProps } from 'react';

interface FileUploadProps extends HTMLProps<HTMLDivElement> {
  error?: string;
  hint?: string;
  children: React.ReactNode;
}

const FileUpload: React.FC<FileUploadProps> = ({
  error,
  hint,
  children,
  ...rest
}: FileUploadProps) => {
  return (
    <div className={`nhsuk-form-group ${error ? 'nhsuk-form-group--error' : ''}`} {...rest}>
      <label htmlFor="file-upload-label" className="nhsuk-label">
        {children}
      </label>
      {error && (
        <span
          id="file-select-error-id"
          data-test-id="file-select-error"
          className="nhsuk-error-message"
        >
          <span className="nhsuk-visually-hidden">Error: </span>
          {error}
        </span>
      )}
      {hint && <Hint data-test-id="file-hint">{hint}</Hint>}
      <input
        aria-describedby="file-upload-label"
        id="file-upload"
        className="nhsuk-file-upload"
        type="file"
      />
    </div>
  );
};

export default FileUpload;
