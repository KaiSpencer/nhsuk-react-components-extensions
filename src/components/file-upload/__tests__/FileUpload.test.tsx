import React from 'react';
import { shallow } from 'enzyme';
import FileUpload from '../FileUpload';

describe('FileUpload', () => {
  it('Matches snapshot', () => {
    const component = shallow(<FileUpload>Upload</FileUpload>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  it('With Error', () => {
    const component = shallow(<FileUpload error="something wrong">Upload</FileUpload>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  it('With Hint', () => {
    const component = shallow(<FileUpload hint="Format: JPG">Upload</FileUpload>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
