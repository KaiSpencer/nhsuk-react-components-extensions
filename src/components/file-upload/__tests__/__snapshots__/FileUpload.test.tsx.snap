// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload Matches snapshot 1`] = `
<div
  className="nhsuk-form-group "
>
  <label
    className="nhsuk-label"
    htmlFor="file-upload-label"
  >
    Upload
  </label>
  <input
    aria-describedby="file-upload-label"
    className="nhsuk-file-upload"
    id="file-upload"
    type="file"
  />
</div>
`;

exports[`FileUpload With Error 1`] = `
<div
  className="nhsuk-form-group nhsuk-form-group--error"
>
  <label
    className="nhsuk-label"
    htmlFor="file-upload-label"
  >
    Upload
  </label>
  <span
    className="nhsuk-error-message"
    data-test-id="file-select-error"
    id="file-select-error-id"
  >
    <span
      className="nhsuk-visually-hidden"
    >
      Error: 
    </span>
    something wrong
  </span>
  <input
    aria-describedby="file-upload-label"
    className="nhsuk-file-upload"
    id="file-upload"
    type="file"
  />
</div>
`;

exports[`FileUpload With Hint 1`] = `
<div
  className="nhsuk-form-group "
>
  <label
    className="nhsuk-label"
    htmlFor="file-upload-label"
  >
    Upload
  </label>
  <Hint
    data-test-id="file-hint"
  >
    Format: JPG
  </Hint>
  <input
    aria-describedby="file-upload-label"
    className="nhsuk-file-upload"
    id="file-upload"
    type="file"
  />
</div>
`;
